solutions/ac-MultipleOf2019.mdx

```diff
@@ -2,11 +2,33 @@
 id: ac-MultipleOf2019
 source: AtCoder Beginner Contest
 title: Multiple of 2019
-author: Maggie Liu, Brad Ma, Mohammad Nour Massri
+author: Maggie Liu, Brad Ma, Mohammad Nour Massri, David Zhang
 ---
 
 [Official Analysis](https://img.atcoder.jp/abc164/editorial.pdf#page=10)
 
+## Explanation
+Firstly, in order to store sections of the string, we can make a suffix array with the condition that at index $$i$$ in suffix array $${S}$$
+
+$${S_i = s_n + 10*s_{n-1} +100*s_{n-2} + ... + {10^{n-i}} * s_{i}}$$
+
+For example, in the string "1234," the suffix array would be
+
+[1234, 234, 34, 4]
+
+This is useful because in $\mathcal{O}(1)$ time, when we use a range query, we can capture any substring with just two indices $$i$$ and $$j$$. So if $${S_0} = 1234 $$ and $${S_3=4}$$, this means $${S_0}-{S_3} = 123 * {10^1}$$ or $$1230$$. 
+
+It also means that if both $${S_0}(mod$$ $$M)=$$ $${S_3}(mod$$ $$M)$$, subtracting would lead to $${S_0}(mod$$ $$M)-$$ $${S_3}(mod$$ $$M)=0$$, meaning a substring that makes a number that is divisible by $$M$$ or $$2019$$. 
+
+So then we can use $${N \choose K}$$ choose formula, and choose 2 numbers $$(2)$$ that are divisible by the same reminder of 2019 $$(N)$$. 
```

Comment:
?

Reply:
got it

Reply:
still, what does this mean

Reply:
If two numbers have that same modulus, like for example 5 (mod 3) and 2(mod 3), subtracting them would equal to a number divisible by the mod or 3. 5-2=3 which is divisible by 3. Same idea here. We can make any number just using two prefixes. And if two prefixes both have the same modulus, then subtracting them from each other should make a number divisible by the mod.

Reply:
don't explain it to me, explain it to the users lol

Reply:
np, doing that rn